### PR TITLE
Remove calling of the old resource in reset()

### PR DESCRIPTION
**Pull request by komamitsu**
For https://github.com/msgpack/msgpack-java/issues/173.

### CODE REVIEW

**Comment by komamitsu** [comment]
@xerial Can you review this PR?

File: msgpack-core/src/main/java/org/msgpack/core/MessagePacker.java [hunk 1]
```diff
@@ -83,20 +83,17 @@ public MessagePacker(MessageBufferOutput out, MessagePack.Config config) {
         this.position = 0;
     }
 
+    /**
+     * Reset output. This method doesn't close the old resource.
+     * @param out new output
+     */
     public void reset(MessageBufferOutput out) throws IOException {
```
Comment (xerial) [inline msgpack-core/src/main/java/org/msgpack/core/MessagePacker.java]: How about returning the old MessageBufferOutput instance here?

File: msgpack-core/src/main/java/org/msgpack/core/MessageUnpacker.java [hunk 2]
```diff
@@ -146,24 +146,21 @@ public MessageUnpacker(MessageBufferInput in, MessagePack.Config config) {
         this.config = checkNotNull(config, "Config");
     }
 
+    /**
+     * Reset input. This method doesn't close the old resource.
+     * @param in new input
+     */
     public void reset(MessageBufferInput in) throws IOException {
```
Comment (xerial) [inline msgpack-core/src/main/java/org/msgpack/core/MessageUnpacker.java]: The same as above: returning the previous MessageBufferInput

File: msgpack-core/src/main/java/org/msgpack/core/buffer/ArrayBufferInput.java [hunk 3]
```diff
@@ -24,6 +24,10 @@ public ArrayBufferInput(byte[] arr, int offset, int length) {
         this.buffer = MessageBuffer.wrap(checkNotNull(arr, "input array is null")).slice(offset, length);
     }
 
+    /**
+     * Reset buffer. This method doesn't close the old resource.
+     * @param buf new buffer
+     */
     public void reset(MessageBuffer buf) {
```
Comment (xerial) [inline msgpack-core/src/main/java/org/msgpack/core/buffer/ArrayBufferInput.java]: `public MesageBuffer reset(MessageBuffer buf)` ?

File: msgpack-core/src/main/java/org/msgpack/core/buffer/ByteBufferInput.java [hunk 4]
```diff
@@ -16,6 +16,10 @@ public ByteBufferInput(ByteBuffer input) {
         this.input = checkNotNull(input, "input ByteBuffer is null");
     }
 
+    /**
+     * Reset buffer. This method doesn't close the old resource.
+     * @param input new buffer
+     */
     public void reset(ByteBuffer input) {
```
Comment (xerial) [inline msgpack-core/src/main/java/org/msgpack/core/buffer/ByteBufferInput.java]: `public ByteBuffer reset(ByteBuffer input)`?

**Comment by xerial** [comment]
@komamitsu 
Do you think it is useful to return the previous MessageBuffer etc. as the return value of `reset(...)` call? 

If not, simply using void is OK.

**Comment by komamitsu** [comment]
@xerial Thanks. I fixed them again. Can you review it again?

**Comment by xerial** [comment]
looks good. Thanks!

**Pull request merged**